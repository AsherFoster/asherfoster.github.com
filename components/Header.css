header {
    z-index: 1;
    position: absolute;
    display: flex;
    width: 100vw;
}
header a {
    padding: 18px;
}
