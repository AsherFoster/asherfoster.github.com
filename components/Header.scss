@import '../lib/_vars';

.nav.mainHeader {
    display: flex;
}
.nav.absolute {
    z-index: 1;
    width: 100vw;
    position: absolute;
}
.nav a {
    color: $text-dark;
    text-decoration: none;
    display: inline-block;
    padding: 10px 12px;
    margin: 8px 6px;
}
.nav a:hover {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 5px;
}

.headerToggle {
    @include buttonReset();
    display: flex;
    align-items: center;
    justify-content: center;
    height: 40px;
    width: 40px;
    right: 10px;
    top: 10px;
    z-index: 90;
    background: $bg-dark;
    border-radius: 20px;
    box-shadow: 1px 2px 10px -1px rgba(0,0,0,0.4);
}
.headerToggle.fixed {
    position: fixed;
}
.headerToggle img, .headerClose img {
    width: 24px;
}

.mobileHeader {
    position: fixed;
    top: 0;
    background: linear-gradient(180deg, rgba(0,0,0,0.90) 400px, rgba(0,0,0,0.7) 100%);
    height: 100vh;
    width: 100vw;
    text-align: center;
    z-index: 90;
}

.headerClose {
    @include buttonReset();
    position: absolute;
    top: 18px;
    right: 18px;
    z-index: 91;
}
.mobileHeader ul {
    padding-left: 0;
    margin-top: 60px;
}
.mobileHeader li {
    color: $text-dark;
    list-style: none;
    font-size: 30px;
    margin: 20px 0;
}

.mobileHeader a {
    text-decoration: none;
}

