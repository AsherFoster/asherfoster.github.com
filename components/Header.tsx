import Link from 'next/link';

const Header = () => (
  <header>
    <Link href='/'>
      <a className='nohover'>Asher Foster</a>
    </Link>
    <span className='flex'/>
    <Link href='/'>
      <a className='header-link'>Home</a>
    </Link>
    <Link href='/portfolio'>
      <a className='header-link'>Portfolio</a>
    </Link>
    <Link href='/contact'>
      <a className='header-link'>Contact</a>
    </Link>
  </header>
);

export default Header;
