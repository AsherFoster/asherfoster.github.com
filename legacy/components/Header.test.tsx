import React from 'react';
import Header from './Header';
import renderer from 'react-test-renderer';

test('Header isn\'t borked', () => {
  const component = renderer.create(
    <Header />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Name doesn\'t show when hidden', () => {
  const component = renderer.create(
    <Header hideName />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
