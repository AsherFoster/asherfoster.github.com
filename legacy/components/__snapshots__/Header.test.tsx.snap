// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header isn't borked 1`] = `
<nav
  className=""
>
  <button
    aria-label="Toggle menu"
    className="undefined mobile"
    onClick={[Function]}
  >
    <img
      alt=""
      src="/static/icons/menu-white.png"
    />
  </button>
  <ul>
    <li>
      <a
        className="nounderline"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        asher foster
      </a>
    </li>
    <span
      className="flex"
    />
    <li>
      <a
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        about
      </a>
    </li>
    <li>
      <a
        href="/portfolio"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        portfolio
      </a>
    </li>
    <li>
      <a
        href="/experimentlist"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        experiments
      </a>
    </li>
    <li>
      <a
        href="/contact"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        contact
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Name doesn't show when hidden 1`] = `
<nav
  className=""
>
  <button
    aria-label="Toggle menu"
    className="undefined mobile"
    onClick={[Function]}
  >
    <img
      alt=""
      src="/static/icons/menu-white.png"
    />
  </button>
  <ul>
    <span
      className="flex"
    />
    <li>
      <a
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        about
      </a>
    </li>
    <li>
      <a
        href="/portfolio"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        portfolio
      </a>
    </li>
    <li>
      <a
        href="/experimentlist"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        experiments
      </a>
    </li>
    <li>
      <a
        href="/contact"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        contact
      </a>
    </li>
  </ul>
</nav>
`;
